.github/workflows/check.yaml

```diff
@@ -22,16 +22,29 @@ jobs:
       tox-version: '<4'
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    name: Functional tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
     needs: lint-unit
     strategy:
       fail-fast: false
       matrix:
-        series: ["focal", "jammy"]
-    with:
-      command: FUNC_ARGS="--series ${{ matrix.series }}" make functional
-      juju-channel: '3.1/stable'
-      nested-containers: true
-      python-version: '3.10'
-      timeout-minutes: 120
-      tox-version: '<4'
+        series: ['focal', 'jammy']
+        juju-channel: ['3.4/stable']
+    steps:
```

Comment:
After some discussion, we agreed that for now we should not move out from bootstack-actions. Can you use them here instead? Thanks.